cachito/workers/tasks/gomod.py

```diff
@@ -16,45 +17,105 @@
+
+    return invalid_gomod_files
+
+
 @app.task
-def fetch_gomod_source(request_id, dep_replacements=None):
+def fetch_gomod_source(request_id, dep_replacements=None, package_configs=None):
     """
     Resolve and fetch gomod dependencies for a given request.
 
     :param int request_id: the Cachito request ID this is for
-    :param list dep_replacements: dependency replacements with the keys "name" and "version"
+    :param list dep_replacements: dependency replacements with the keys "name" and "version"; only
+        supported with a single path
+    :param list package_configs: the list of optional package configurations submitted by the user
     :raises CachitoError: if the dependencies could not be retrieved
     """
     config = get_worker_config()
+    if package_configs is None:
+        package_configs = []
+
     bundle_dir = RequestBundleDir(request_id)
+    subpaths = [os.path.normpath(c["path"]) for c in package_configs if c.get("path")]
+
+    if not subpaths:
+        # Default to the root of the application source
+        subpaths = [os.curdir]
 
-    if not bundle_dir.go_mod_file.exists():
-        if config.cachito_gomod_ignore_missing_gomod_file:
-            log.warning("The go.mod file is missing for request %d", request_id)
+    invalid_gomod_files = _find_missing_gomod_files(bundle_dir, subpaths)
+    if invalid_gomod_files:
+        invalid_files_print = "; ".join(invalid_gomod_files)
+        file_suffix = "s" if len(invalid_gomod_files) > 1 else ""
+
+        # missing gomod files is supported if there is only one path referenced
+        if config.cachito_gomod_ignore_missing_gomod_file and len(subpaths) == 1:
+            log.warning("go.mod file missing for request at %s", invalid_files_print)
             return
 
-        raise CachitoError("The go.mod file is missing")
-
-    log.info("Fetching gomod dependencies for request %d", request_id)
-    request = set_request_state(request_id, "in_progress", "Fetching the gomod dependencies")
-    try:
-        gomod = resolve_gomod(str(bundle_dir.source_dir), request, dep_replacements)
-    except CachitoError:
-        log.exception("Failed to fetch gomod dependencies for request %d", request_id)
-        raise
-
-    env_vars = {
-        "GOCACHE": {"value": "deps/gomod", "kind": "path"},
-        "GOPATH": {"value": "deps/gomod", "kind": "path"},
-    }
-    env_vars.update(config.cachito_default_environment_variables.get("gomod", {}))
-    update_request_with_package(request_id, gomod["module"], env_vars)
-    update_request_with_deps(request_id, gomod["module"], gomod["module_deps"])
-
-    # add package deps
-    for package in gomod["packages"]:
-        if package.get("pkg_deps"):
-            # This also adds the package to the request
-            update_request_with_deps(request_id, package["pkg"], package["pkg_deps"])
+        raise CachitoError(
+            "The {} file{} must be present for the gomod package manager".format(
```

Comment:
Nit pick (If applicable)
```suggestion
            "The {} file {} must be present for the gomod package manager".format(
```

Reply:
The string that gets interpolated here is either `""` or `"s"` (file vs. files), so this is intended